src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/casenotes/CaseNotesReconciliationServiceTest.kt

```diff
@@ -0,0 +1,147 @@
+package uk.gov.justice.digital.hmpps.prisonertonomisupdate.casenotes
+
+import com.microsoft.applicationinsights.TelemetryClient
+import kotlinx.coroutines.test.runTest
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.casenotes.model.CaseNote
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.nomissync.model.CaseNoteAmendment
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.nomissync.model.CaseNoteResponse
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.nomissync.model.CodeDescription
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.nomissync.model.PrisonerCaseNotesResponse
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.nomissync.model.PrisonerId
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.NomisApiService
+import java.time.LocalDateTime
+
+private const val OFFENDER_NO = "A3456GH"
+private const val DPS_CASE_NOTE_ID = "12345678-0000-0000-0000-000011112222"
+
+class CaseNotesReconciliationServiceTest {
+
+  private val caseNotesApiService: CaseNotesDpsApiService = mock()
+  private val caseNotesNomisApiService: CaseNotesNomisApiService = mock()
+  private val nomisApiService: NomisApiService = mock()
+  private val telemetryClient: TelemetryClient = mock()
+
+  val nomisPrisoner = PrisonerCaseNotesResponse(
+    caseNotes = listOf(
+      CaseNoteResponse(
+        caseNoteId = 1L,
+        bookingId = 1L,
+        caseNoteType = CodeDescription("CODE", "desc"),
+        caseNoteSubType = CodeDescription("SUBCODE", "desc"),
+        authorStaffId = 101L,
+        authorUsername = "USER",
+        authorLastName = "SMITH",
+        caseNoteText = "the actual text",
+        amendments = listOf(
+          CaseNoteAmendment(
+            text = "the amendment text",
+            authorUsername = "AMUSER",
+            createdDateTime = "2024-01-01T01:02:03",
+          ),
+        ),
+        occurrenceDateTime = "2024-01-01T01:02:03",
+        createdDatetime = "notused",
+        createdUsername = "notused",
+      ),
+    ),
+  )
+
+  fun dpsPrisoner(
+    type: String = "CODE",
+    amendmentText: String = "the amendment text",
+    amendments: List<uk.gov.justice.digital.hmpps.prisonertonomisupdate.casenotes.model.CaseNoteAmendment> = listOf(
```

Comment:
I think this type is already imported

Reply:
Its the nomis type of the same name that is imported

Reply:
Then it might be nicer to also import this and rename it using `as`